### ISSUE TITLE
pipewire: pw_stream_update_properties message whenever track changes

### ISSUE THREAD

**Issue by elomatreb** [MusicPlayerDaemon/MPD#1796]
### Describe the bug

When using the Pipewire output plugin, the following message is logged whenever the currently playing track changes:

```
*** pw_stream_update_properties called from wrong context, check thread and locking: Operation not permitted
```

## Expected Behavior

No  error is logged.

## Actual Behavior

See above.

## Version

<details>
<summary>MPD Version (0.23.12)</summary>

```
Music Player Daemon 0.23.12 (0.23.12)
Copyright 2003-2007 Warren Dukes <warren.dukes@gmail.com>
Copyright 2008-2021 Max Kellermann <max.kellermann@gmail.com>
This is free software; see the source for copying conditions.  There is NO
warranty; not even MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Database plugins:
 simple proxy upnp

Storage plugins:
 local udisks nfs curl

Neighbor plugins:
 upnp udisks

Decoders plugins:
 [mad] mp3 mp2
 [mpg123] mp3
 [vorbis] ogg oga
 [oggflac] ogg oga
 [flac] flac
 [opus] opus ogg oga
 [sndfile] wav aiff aif au snd paf iff svx sf voc w64 pvf xi htk caf sd2
 [audiofile] wav au aiff aif
 [dsdiff] dff
 [dsf] dsf
 [hybrid_dsd] m4a
 [faad] aac
 [mpcdec] mpc
 [wavpack] wv
 [openmpt] mptm mod s3m xm it 669 amf ams c67 dbm digi dmf dsm dtm far imf ice j2b m15 mdl med mms mt2 mtm nst okt plm psm pt36 ptm sfx sfx2 st26 stk stm stp ult wow gdm mo3 oxm umx xpk ppm mmcmp
 [modplug] 669 amf ams dbm dfm dsm far it med mdl mod mtm mt2 okt s3m stm ult umx xm
 [mikmod] amf dsm far gdm imf it med mod mtm s3m stm stx ult uni xm
 [sidplay] sid mus str prg P00
 [wildmidi] mid
 [fluidsynth] mid
 [ffmpeg] 16sv 3g2 3gp 4xm 8svx aa3 aac ac3 adx afc aif aifc aiff al alaw amr anim apc ape asf atrac au aud avi avm2 avs bap bfi c93 cak cin cmv cpk daud dct divx dts dv dvd dxa eac3 film flac flc fli fll flx flv g726 gsm gxf iss m1v m2v m2t m2ts m4a m4b m4v mad mj2 mjpeg mjpg mka mkv mlp mm mmf mov mp+ mp1 mp2 mp3 mp4 mpc mpeg mpg mpga mpp mpu mve mvi mxf nc nsv nut nuv oga ogm ogv ogx oma ogg omg opus psp pva qcp qt r3d ra ram rl2 rm rmvb roq rpl rvc shn smk snd sol son spx str swf tak tgi tgq tgv thp ts tsp tta xa xvid uv uv2 vb vid vob voc vp6 vmd wav webm wma wmv wsaud wsvga wv wve rtp:// rtsp:// rtsps://
 [gme] ay gbs gym hes kss nsf nsfe rsn sap spc vgm vgz
 [pcm]

Filters:
 libsamplerate soxr

Tag plugins:
 id3tag

Output plugins:
 shout null fifo pipe alsa ao oss openal solaris pipewire pulse jack httpd snapcast recorder

Encoder plugins:
 null vorbis opus lame twolame wave flac

Archive plugins:
 [bz2] bz2
 [zzip] zip
 [iso] iso

Input plugins:
 file io_uring archive alsa qobuz curl ffmpeg nfs mms cdio_paranoia

Playlist plugins:
 extm3u m3u pls xspf asx rss soundcloud flac cue embcue

Protocols:
 file:// alsa:// cdda:// ftp:// ftps:// gopher:// hls+http:// hls+https:// http:// https:// mms:// mmsh:// mmst:// mmsu:// nfs:// qobuz:// rtmp:// rtmpe:// rtmps:// rtmpt:// rtmpte:// rtmpts:// rtp:// rtsp:// rtsps:// scp:// sftp:// smb:// srtp://

Other features:
 avahi dbus udisks epoll icu inotify ipv6 systemd tcp un
```
</details>

I'm using Pipewire 0.3.70 with Wireplumper 0.4.14.

## Configuration
<details>
<summary>Configuration</summary>

```
music_directory "~/Music/Shared"
playlist_directory "~/.config/mpd/playlists"

db_file "~/.cache/mpd/database"
state_file "~/.cache/mpd/state"
sticker_file "~/.cache/mpd/sticker"

bind_to_address "localhost:6600"

restore_paused "yes"

input {
    plugin "curl"
}

audio_output {
    type "pipewire"
    name "pipewire"
}

replaygain "album"
max_output_buffer_size "16384"
zeroconf_enabled "no"

decoder {
    plugin "wildmidi"
    enabled "no"
}
```
</details

**Comment by MaxKellermann** [comment]
Was fixed 2 months ago by eabc487e597dd326c2cb67f33cfef073aafeaf59

**Comment by irishmonk-57** [comment]
May I ask if this "bug" would still be in mpd version 0.23.12? That's the version that I have on my Debian system ('bookworm stable'), and I am seeing this error in `journalctl`: 

```
mpd[19842]: player: played "Loch Lomond [GXGVFJqSqqg].m4a"
mpd[19842]: *** pw_stream_update_properties called from wrong context, check thread and locking: Operation not permitted
```

**Comment by MaxKellermann** [comment]
Of course it is. 0.23.12 was released on 2023-01-17 and the bug fix was made on 2023-03-06.

Do not use outdated MPD versions! And no, "but... Debian!" is a lousy excuse. https://www.musicpd.org/download-unoff-debian/

**Comment by irishmonk-57** [comment]
> Of course it is. 0.23.12 was released on 2023-01-17 and the bug fix was made on 2023-03-06.

"Of course..." How would one know the release date of 0.23.12? It's not included in `mpd --version`; it's not easy to find from this GitHub site - at least I don't see how to do it (after trying). So I am curious why you preface your remark with *"Of course..."*. If I've overlooked something, please do let me know. 

> Do not use outdated MPD versions! And no, "but... Debian!" is a lousy excuse. https://www.musicpd.org/download-unoff-debian/ 

Debian is the most widely used Linux distribution in the world. Yes - they are slow to update packages, but hopefully you can see where this can be advantageous for stability. Perhaps it would be in your interest to work more closely with Debian? Perhaps focus on reliability/stability instead of more and newer features? 

In conclusion, I'd like to share this with you: Having used Linux for "a few years", and having tried most of the (cmd line) music players, I'm convinced that `mpd` is a "better idea". And I appreciate the fact that you are investing quite a bit of your time in improving it.  If there's anything I can do to help, please let me know.

**Comment by MaxKellermann** [comment]
> "Of course..." How would one know the release date of 0.23.12? It's not included in `mpd --version`; it's not easy to find from this GitHub site - at least I don't see how to do it (after trying). So I am curious why you preface your remark with _"Of course..."_. If I've overlooked something, please do let me know.

Every copy of MPD comes with the changelog which includes the release dates of all MPD versions for the last 22 years. Or you can look in the git repository at the release tags. That's what you overlooked.

> Debian is the most widely used Linux distribution in the world. Yes - they are slow to update packages, but hopefully you can see where this can be advantageous for stability. Perhaps it would be in your interest to work more closely with Debian? Perhaps focus on reliability/stability instead of more and newer features?

I do. Did you even click on the link?

If you don't use the latest MPD version (e.g. from that link) and instead use the version that comes with Debian stable, you won't get any support from this project. If you don't like that, don't use Debian or don't use MPD (or just use the stuff behind the link). But don't complain or ask anybody else to do something. It's up to you, and you don't pay for support.

**Comment by irishmonk-57** [comment]
> Every copy of MPD comes with the changelog which includes the release dates of all MPD versions for the last 22 years. Or you can look in the git repository at the release tags. That's what you overlooked.
> 

Ahh! the "git repository at the release tags" ...  yes, my apologies.

> > Debian is the most widely used Linux distribution in the world. Yes - they are slow to update packages, but hopefully you can see where this can be advantageous for stability. Perhaps it would be in your interest to work more closely with Debian? Perhaps focus on reliability/stability instead of more and newer features?
> 
> I do. Did you even click on the link?
> 
> If you don't use the latest MPD version (e.g. from that link) and instead use the version that comes with Debian stable, you won't get any support from this project. If you don't like that, don't use Debian or don't use MPD (or just use the stuff behind the link). But don't complain or ask anybody else to do something. It's up to you, and you don't pay for support.

Of course I did, but was hesitant to add an unknown source. I've not lost the reference though - I may use it later.  

In the meantime, one of the maintainers was motivated to move `ver 0.24.2` to backports, and I'm using that. I've not used it long enough to form an opinion, but it seems to be more reliable. And looking at the `journalctl` logs just now, I'm not seeing those odd `I am done with this track` messages from `lib mpg`. 

WRT your ultimatums: No - I don't pay for support, but reminding me of that seems pointless. If you want to run your project that way, I'm sure that is entirely up to you. Are you mad at me?... at your users in general?... or what? There's really no need to get testy about it, eh?
